Bela/main.cpp

```diff
@@ -199,6 +203,71 @@ struct DigiIO : csnd::InPlug<3> {
   }
 };
 
+/*  TrillIn opcode 
+  read values from Trill sensors
+  array sizes must be same as inumTouches
+  kactiveTouches, ksizes[], kvertLocations[], khoriLocations[] trill inumTouches, itrillID
+*/
+struct TrillIn : csnd::Plugin<4,2> {
+  int activeTouches = 0;
+  int numTouches;
+  int trillID;
+  float *touchSize; 
+  float *touchVertLocation;
+  float *touchHoriLocation;
+  
+  int init() {
+    numTouches = inargs[0];
+    trillID = inargs[1];
+    if (gTouchSensors.size() < trillID+1)
+      return csound->init_error("No Trill sensor connected");
+
+    touchSize = new float[numTouches];
+    touchVertLocation = new float[numTouches];
+    touchHoriLocation = new float[numTouches];    
+    return OK;
+  }
+  
+  int kperf () {
+    activeTouches = gTouchSensors[trillID]->getNumTouches();
+    outargs[0] = MYFLT (activeTouches);
+    // Read locations from Trill sensor
+    for(unsigned int i = 0; i < activeTouches; i++) {
```

Comment:
this only works when the devices are in `Centroid` mode, which is the default mode for all devices except Trill Craft. For Trill Craft you should probably read `gTouchSensors[trillId].rawData[n]` instead. Then below in the loop you could differentiate between 1D and 2D devices. See [here](https://github.com/BelaPlatform/Bela/blob/dev/core/default_libpd_render.cpp#L900-L927) for an example

Reply:
Maybe I should make a trillCraft opcode for the Trill Craft? I've only had a chance to test Ring, Square and Bar. I also realised even the Square might have its specific opcode to be able to vary between separate touches and compoundtouch. Or I guess that could be a general argument. I'll have a look at it for the next PR